CMakeLists.txt

```diff
@@ -636,10 +636,12 @@ set(ZIG_STD_FILES
     "special/build_runner.zig"
     "special/builtin.zig"
     "special/compiler_rt.zig"
+    "special/compiler_rt/arm/aeabi_fcmp.zig"
```

Comment:
I don't know if we need a new directory for arm.

Reply:
I'm following the directory structure of compiler-rt.

Reply:
Opinion: up until now we don't have sub-directories, so I think it should be flat, unless re run into namespace conflicts.

Reply:
I tend to like flat directory structures too. It's already pretty nested, but nesting it makes the intention to follow compiler-rt more explicit.

I would like to consider that we don't follow compiler-rt exactly. The hard floating point targets can just have the hard floating point instructions directly in the functions that provide them instead of branching all over the place to functions from other files. Compiler-rt has a complicated (seemingly to me) system for determining which version of a function gets linked. I think compiler-rt is a great starting point, but if we are already doing a port, I would like to be able to reorder/insert inline some stuff. That would make the port for hard float ABI progress at the same time as the soft float ABI. Of course that does make it harder to compare against future releases of compiler-rt.